README.md

```diff
@@ -305,6 +306,37 @@ method takes one argument:
 
 2. `chain` must return a value of the same Chain
 
+### ChainRec
+
+A value that implements the ChainRec specification must also implement the Chain specification.
+
+1. ```js
+   m.chainRec((next, done, v) => p(v) ? d(v).map(done) : n(v).map(next), i)
+   ```
+   is equivalent to
+
+   ```js
+   (function step(v) { return p(v) ? d(v) : n(v).chain(step); }(i))
+   ```
+   But stack usage of `m.chainRec(f, i)` must be at most a constant multiple of the stack usage of `f` itself. (equivalence)
+
+#### `chainRec` static method
+
+A Type which has a ChainRec must provide an `chainRec` method on itself
+or its `constructor` object. The `chainRec` method takes two arguments:
+
+    a.chainRec(f, i)
+    a.constructor.chainRec(f, i)
+
+1. `f` must be a function which returns a value
+   1. If `f` is not a function, the behaviour of `chainRec` is unspecified.
+   2. `f` takes three arguments `next`, `done`, `value`
+       `next` and `done` are functions which take one argument and return some value.
+   3. `f` must return a value of the same ChainRec which contains a value returned from either `done` or `next`
+   4. The third argument to `f` on it's first invocation should be `i`
```

Comment:
Maybe instead of this we should state that:

- The third argument to `f` must be of same type as `i`
- The first argument of `next` must be of same type as `i`

Basically mirror type signature `((a -> c) -> (b -> c) -> a -> m c) -> a -> m b`